versioned/tiered/src/main/java/com/dremio/nessie/tiered/builder/L1Builder.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.tiered.builder;
+
+import com.dremio.nessie.versioned.Key;
+import com.dremio.nessie.versioned.store.Id;
+
+/**
+ * Interface to create an L1 Builder. To be implemented by each {@link Store} implementation.
+ *
+ * @param <T> The concrete type built by this builder.
+ */
+public interface L1Builder<T> {
+
+  /**
+   * Construct the expected object.
+   * @return The built object.
+   */
+  T build();
+
+  L1Builder<T> commitMetadataId(Id id);
+
+  IncrementalKeyListBuilder<T> incrementalKeyList();
+
+  CompleteKeyListBuilder<T> completeKeyList();
+
+  ParentListBuilder<T> parentList();
```

Comment:
Seems that each `L1` has a single list of parents and each parent is defined by exactly one `Id`. Wonder if we really need a `l1builder.parentList().addAncestor(parentId).build()` instead of just a `l1builder.addAncestor(parentId)`. I.e. remove the `ParentListBuilder` interface and move it's only method here.